lib/mint/http1.ex

```diff
@@ -206,7 +212,8 @@ defmodule Mint.HTTP1 do
         state: :open,
         log: log?,
         case_sensitive_headers: Keyword.get(opts, :case_sensitive_headers, false),
-        skip_target_validation: Keyword.get(opts, :skip_target_validation, false)
+        skip_target_validation: Keyword.get(opts, :skip_target_validation, false),
+        optional_responses: Keyword.get(opts, :optional_responses, [])
       }
```

Comment:
instead of storing a list of potentially invalid values, my suggestion would be to validate them as they come in, perhaps something like this:

```suggestion
      }
      |> parse_optional_responses(Keyword.get(opts, :optional_responses, []))
```

```elixir
defp parse_optional_responses(conn, [:status_reason | rest]) do
  parse_optional_responses(%{conn | status_reason: true}, rest)
end

defp parse_optional_responses(conn, []) do
  conn
end

defp parse_optional_responses(conn, [invalid | _]) do
  raise ArgumentError, """
  invalid :optional_responses value #{inspect(value)}, allowed values are:

    * :status_reason
  """
end
```